package xia;


enum XSocketMsgType {
  XSOCKET_OPEN = 0;
  XSOCKET_BIND = 1;
  XSOCKET_CLOSE = 2;
  XSOCKET_CONNECT = 3;
  XSOCKET_ACCEPT = 4;
  
  XSOCKET_PUTCID = 5;
  XSOCKET_SENDTO = 6;
  XSOCKET_DATA = 7;

  XSOCKET_REQUEST_FAILED = 8;
}


message XSocketMsg {
  required XSocketMsgType type = 1; // XIA Socket message type
  
  optional string sdag = 2; // Source DAG XIP
  optional string ddag = 3; // Destination DAG XIP
  optional bytes payload = 4; // data  
}



message msg {
  optional int32 appid = 1;  // port number?
  optional bytes xid = 2;
  optional string xiapath_src = 5;
  optional string xiapath_dst = 6;
  optional bytes payload = 3;  
  enum MsgType {
    GETLOCALHID = 0;
    GETCID = 1;
    CONNECTSID = 2;
    PUTCID = 3;
    SERVESID = 4;
  }
  optional MsgType type = 4;

}

message msg_response {
  required int32 appid = 1;
  repeated bytes xid = 2;   // can be a list of CID's
  optional string payload = 3;  // bytes or string?
}
